client/src/pages/Member.js

```diff
@@ -1,5 +1,147 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { Alert, Icon } from '@hack4impact-uiuc/bridge';
+import PropTypes from 'prop-types';
+import StringAttribute from '../components/EditableAttribute/StringAttribute';
+import EnumAttribute from '../components/EditableAttribute/EnumAttribute';
+import {
+  getMemberByID,
+  getMemberEnumOptions,
+  getMemberPermissionsByID,
+  getMemberSchemaTypes,
+} from '../utils/apiWrapper';
+import BooleanAttribute from '../components/EditableAttribute/BooleanAttribute';
+import DateAttribute from '../components/EditableAttribute/DateAttribute';
 
-const Member = () => <div />;
+const areResponsesSuccessful = (...responses) => {
+  let success = true;
+  responses.forEach(response => {
+    if (response == null || response.data == null || !response.data.success)
+      success = false;
+  });
+
+  return success;
+};
+
+const Member = ({ memberID }) => {
```

Comment:
As the project is structured right now, we access the Member Page through a router redirect to the endpoint **{BASE_URL}/member**. This means that in the table when we have a button to send a user to that members profile page it will access App.js and render the Member page that way. I think the problem with this approach is  that we would end up sending the same memberID every single time because we have no way of getting the clicked member's id to the component. 

### Idea
Modify the router path for /member to include **/member/:id**. Then either:
1. Call a function within App.js during the router call to get the member id and pass it through the props of the Member component. 
2. Don't pass the member id as props into the component. Instead, 
```
const [memberID, setMemberID] = useState(null);

  /**
   * Parases/Returns the member ID from the URL
   * @param {String} url the current webpage url
   */
  const parseMemberID = (url) => {
    const urlPartsArray = url.split("/"); // [BASE_URL, member, :memberID]
    // we don't care about array integrity here
    const id = urlPartsArray.pop(); // removes the last element from array
    return id;
  }
  useEffect(() => {
    setMemberID(parseMemberID(window.location.href)); // could also use useLocation Hook here
  }, []);
```
I think option 2 may be a bit better because processing the url parsing during the router redirection could lead to a bad UX experience with slower redirects

If this has been accounted for in some other way let me know, I just don't see how the table is gonna be able to communicate to the member component what ID the user just clicked on.

Reply:
Yes, we'd definitely need a route change. But we don't have to pass the memberId through props or parse it from the URL. We can use the `useParams` hook from `react-router-dom` to access the `memberId` param. See here: https://reactrouter.com/web/example/url-params

Reply:
We can make this a new PR though, it'd only be a couple lines of code.

Reply:
Those are all good points @amit-sawhney, I think your "option 2" is a pretty clean way of going about this. I'll get all the routing in place. For now, I'm going to make the "View Profile" button always redirect to the url with the mock data in the DB -- once we actually get real member data in there, it will be an easy enough to replace.

Thanks @amit-sawhney and @Qrtn